.github/workflows/database.yml

```diff
@@ -0,0 +1,42 @@
+name: Database CI
+on: push
+jobs:
+  migrations:
+    name: migrations
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:13
+        env:
+          POSTGRES_PASSWORD: rpg
+          POSTGRES_USER: rpg
+          POSTGRES_DB: rpg
+          DB_PORT: 5432
+        ports:
+          - 5432:5432
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Use Node.js 14
+      uses: actions/setup-node@v1
+      with:
+        node-version: 14
+    - name: Cache node modules
+      uses: actions/cache@v2
+      env:
+        cache-name: node-modules
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
+    - name: Install node modules
+      run: npm ci
+    - name: Create postgrator config
+      run: cp postgrator.sample.json postgrator.json
+    - name: Run migrations
+      run: npm run db:migrate
+    - name: Undo migrations
+      run: npm run db:migrate -- 0
```

Comment:
The real interesting part of each job is the `steps` section.

Each step is given a `name`, and either `uses` a public [action](https://github.com/marketplace?type=actions) (or private action you wrote yourself), or just `run` a script directly. Actions are basically just bigger scripts that are useful enough to be worth publishing and sharing, and take various options.

The actions used here:
* [`actions/checkout`](https://github.com/actions/checkout): checks out the repository
* [`actions/setup-node`](https://github.com/actions/setup-node): sets up Node JS (installs Node, NPM, etc)
* [`actions/cache`](https://github.com/actions/cache): caches some files between runs so that we can save time. We use it to cache the `node_modules` so that if they haven't changed we don't need to reinstall, making the time to run the workflow a little quicker. I just copy this part off the [examples](https://github.com/actions/cache/blob/main/examples.md#node---npm).

Then the last 4 steps are just using the things we have set up and running the migrations. The test and lint jobs from the other file look very similar but run test and lint instead of the migrations.

Github decides whether the job was successful or failed based on the result of running each command. If any encounter an error (exit code != 0), then it's a failure and the rest of the job is aborted, otherwise if it reaches the end without any error, then it's a success.